14 create a search component AOA

AOA19 opened the pull request:
## What changes did I make?
- Created the SearchBar component based on the Figma design and passed in the props from the API request in App.jsx
- Deleted the temporary search bar that was inside the App.jsx component 
- Removed styling of the temporary search bar in index.css and replaced it with spacing reset for the whole app
- Added a css file for the SearchBar component 
- Added the react-icons library as a dependency
- Deleted boilerplate styling from App.css
- Deleted .eslintcache file and added to gitignore file 
- Removed "jsx" extension for App import inside of main.jsx


## Why did you make these changes?
- Needed to create a separate search bar component for the app 
- Index.css will be used for global styling only that's why I reset the spacing of the entire app. Any styling for the App.jsx component will go inside of App.css


## Visual changes (if applicable)
- Empty search bar 
<img width="385" alt="search-bar-1" src="https://github.com/chingu-voyages/v46-tier2-team-20/assets/66294921/61898744-08f0-4984-8f4e-6e469a0a3196">

- Search bar with in
<img width="385" alt="search-bar-2" src="https://github.com/chingu-voyages/v46-tier2-team-20/assets/66294921/7d071e96-c52e-4bac-a932-e3ddd5754335">
put

AOA19 committed:
Removed .eslintcache file and added to gitignore

AOA19 committed:
Added react-icons library as a dependency

AOA19 committed:
Removed intial React styling for App.css, added global spacing styling for index.css

AOA19 committed:
Created SearchBar component, added to App component and removed test code from App.jsx

AOA19 committed:
Merged chnages from development branch

AOA19 committed:
Removed temp search bar from App.jsx and passed props to SearchBar component

allison-truhlar commented:
@AOA19 This search component is looking good! I have one edit to make and a couple of suggestions/questions:

To edit:
- I think we need to also have ingredients sent as a prop to the search form, and then inside the form, add a property, value={ingredients}, in order to make the form a controlled input.

Suggestions/questions (do not need to be addressed before merging):
- I think we will need to add some sort of instructions/validation to the user input to ensure each ingredient is separated by a space for searching. We could also consider changing to a separate input per ingredient, as discussed on Discord.
- I see you have an onReset function in the form - I'm not familiar with this. After some Googling, I'm wondering - do we need to add a reset button for this function to work? If not, when is it called?

AOA19 committed:
Merged current changes from developement branch

AOA19 committed:
Change title to name of app

AOA19 committed:
Added validation to search bar, moved ingredient state and search function from App.jsx to SearchBar.jsx

AOA19 commented:
@allison-truhlar, @Emimint, @komalgill0310 
- Instead of passing ingredients as a prop to the search bar I moved the whole useState hook to SearchBar.jsx. It made more sense to have it in the SearchBar component because it was the only component that was using it. So I only pass the fetchData function as a prop to the SearchBar component. 
- I left the recipes and error hooks in App.jsx since other components will be using those hooks. I commented out the isSearching hook because I wasn't sure if we needed it anymore or not.
- I also move the handleSearch function from App.jsx to SearchBar.jsx. It made more sense to write that function inside the search component. And inside that function I added the validation that you suggested.
- An error will display when the search bar is submitted without an input (*see image 1*) and an error will display when the input is an invalid format (*see image 2*), based on the regular expression. 
- The valid format is any letter (not case sensitive), at least 3 characters long, no numbers, required space separating each word and an optional comma. 


**Image 1**
<img width="385" alt="empty-input-error" src="https://github.com/chingu-voyages/v46-tier2-team-20/assets/66294921/0821322c-09c3-4803-b8d3-f4cada824659">


**Image 2**
<img width="385" alt="invalid-format-error" src="https://github.com/chingu-voyages/v46-tier2-team-20/assets/66294921/361f517f-2db6-4fe8-8ae3-ac9cf1b39798">

robester0403 commented on src/App.jsx:
```diff
@@ -37,56 +38,27 @@ function App() {
       if (response.data.count > 0) {
```
You don't need an if statement here. Unless you have you have a specific message to display if it comes back with 0 results.

robester0403 commented on src/App.jsx:
```diff
@@ -5,18 +5,19 @@ import axios from 'axios';
-  // In that case, I think the response or error could be returned, and then used with setRecipes or setError here
+  // In that case, I think the response or error could be returned,
+  // and then used with setRecipes or setError here
   async function fetchData(ingredientString) {
```
This is an api fetching function and should be refactored from this layer. This file does not specifically focus on this layer of logic. It should be exported from an API file.

robester0403 replied:
In addition, you need to reset the error message so the new search clears the error messaging from the previous search.

robester0403 commented on src/App.jsx:
```diff
@@ -5,18 +5,19 @@ import axios from 'axios';
 import Header from './components/header/Header';
+import SearchBar from './components/searchBar/SearchBar';
 
 function App() {
```
You should reserve the App file for more providers and context related wrappers. And wrap them around a main page file that should hold all the logic you have currently on this page.

robester0403 reviewed (commented):
Initial comments

komalgill0310 committed:
Merge branch 'development' into 14-create-a-search-component-AOA
